### ISSUE TITLE
mask parameter in Transformer support 3D tensor but not mentioned

### ISSUE THREAD

**Issue by cloneniu** [pytorch/pytorch#74612]
## 📚 Documentation

(Add a clear and concise description of what the documentation content issue is. A link to any relevant https://pytorch.org page is helpful if you have one.)

First, I read doc in this https://pytorch.org/docs/stable/generated/torch.nn.TransformerEncoderLayer.html page, which redirect me to the doc on this https://pytorch.org/docs/stable/generated/torch.nn.Transformer.html page. The notes about "src_mask" parameter under "forward--shape" section says it only support shape of (S, S). However when I track down the source code I found "src_mask" is actually the "attn_mask" on this https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html page which support 3D tensor. Then I tested feeding 3D mask to TransformerEncoderLayer and it works. Please check.

cc @brianjo @mruberry @jbschlosser @bhosmer @cpuhrsch

**Comment by jbschlosser** [comment]
Thanks for the report, @cloneniu! We'd accept a PR expanding the supported shapes listed in the `nn.Transformer` documentation:
* `src_mask` - supports shape `(S, S)` or `(N * num_heads, S, S)`
* `tgt_mask` - supports shape `(T, T)` or `(N * num_heads, T, T)`
